### PR TITLE
[Enhancement] ternModuleInstalls to support folders within a plug-in JAR per Issue #332

### PR DESCRIPTION
**Pull request by demonfiddler**
Hi Angelo, I finally received legal approval to commit these changes and raise a PR. Let me know what you think. Cheers, --A

P.S. This PR also fixes a couple of minor buglets that I noticed.

### CODE REVIEW

**Comment by angelozerr** [comment]
Thanks a lot @demonfiddler for your PR, but I'm doing a big refactoring with tern repository to use it with debugger (you can now debug with breakpoint tern plugins with JavaScript debugger!).

So please wait that I commit my work to adjust your PR (I'm sorry with that). I have renamed getTernBaseDir with  getTernRepositoryBaseDir ( https://github.com/angelozerr/tern.java/pull/361/files#diff-3dcdb5c3da91a48f25a46d3a7f67c18aR82)

public static File getTernRepositoryBaseDir() throws IOException {
		return FileLocator.getBundleFile(Platform
				.getBundle(ternjs.Activator.PLUGIN_ID));
	}

It's not commited, please wait...

Otherwise I would like to know if you be interested with https://github.com/angelozerr/tern.java/issues/193 ?

**Comment by demonfiddler** [comment]
Okay @angelozerr I'll wait 'til you've completed your refactoring and attempt to merge your changes with mine before raising a new PR.

Issue [#193](https://github.com/angelozerr/tern.java/issues/193) is of (passing) interest, as in our particular case the API in question (defined by some 61 generated *.js files) is static in any given release, so it is appropriate for us to condense it during the Maven build and include only the resulting JSON file in the Eclipse plug-in. We may use the same JSON file with Tern in other environments besides Eclipse, too.

**Comment by angelozerr** [comment]
@demonfiddler I have commited my big refactoring, please update your PR, thanks!

File: eclipse/tern.eclipse.ide.core/src/tern/eclipse/ide/core/TernCorePlugin.java [hunk 1]
```diff
@@ -81,7 +81,8 @@ public static File getTernCoreBaseDir() throws IOException {
 
 	public static File getTernBaseDir() throws IOException {
 		return new File(FileLocator.getBundleFile(Platform
-				.getBundle(ternjs.Activator.PLUGIN_ID)), "node_modules/tern");
+				.getBundle(ternjs.Activator.PLUGIN_ID)), "node_modules/tern")
+				.getCanonicalFile();
```
Comment (angelozerr) [inline eclipse/tern.eclipse.ide.core/src/tern/eclipse/ide/core/TernCorePlugin.java]: Why calling getCanonicalFile(). here? What is the problem you are fixing?

For your information getTernBaseDir was removed and replaced with getTernRepositoryBaseDir (root folder of tern repository).

Comment (demonfiddler) [inline eclipse/tern.eclipse.ide.core/src/tern/eclipse/ide/core/TernCorePlugin.java]: When I stepped through this code in the debugger I could see that the file path was not correctly normalized for the platform or something – either that or it contained a ‘..’ segment. Whatever it was, it was ugly and this change fixed the ugliness.

**Comment by demonfiddler** [comment]
Okay, thanks Angelo, I will look at this next week. Cheers, --A (Adrian Price aka demonfiddler)

**Comment by demonfiddler** [comment]
Closing this PR in order to merge latest from Angelo before re-creating the PR.

**Pull request closed without merging**